### ISSUE TITLE
Altering Resource Monitors doesnt work as expected

### ISSUE THREAD

**Issue by curtisq** [Snowflake-Labs/terraform-provider-snowflake#1832]
**Provider Version**

The provider version you are using.
0.55.1

**Terraform Version**

The version of Terraform you were using when the bug was encountered.

1.0.11

**Describe the bug**

Changing only the triggers of a resource monitor throws a syntax error

For example take the following monitor:
```
resource "snowflake_resource_monitor" "some_monitor" {
  name         = "some_monitor"
  warehouses   = [snowflake_warehouse.some_warehouse.name]
  credit_quota = 10

  frequency       = "WEEKLY"
  start_timestamp = "2023-02-02T00:00:00Z"

  notify_triggers            = [90]
  suspend_triggers           = [100]
  suspend_immediate_triggers = [105]

  notify_users = [snowflake_user.some_user.name]
}
```

If I were to say change a trigger line to `notify_triggers            = [95]` and change nothing else then the provider produces the following SQL:
```
ALTER RESOURCE MONITOR "some_monitor" 
SET  
TRIGGERS ON 100 PERCENT DO SUSPEND 
    ON 105 PERCENT DO SUSPEND_IMMEDIATE 
    ON 95 PERCENT DO NOTIFY
```
This fails with 
```
syntax error line 3 at position 24 unexpected 'DO'.
syntax error line 4 at position 4 unexpected 'ON'.
...
```

**Expected behavior**

Based on the [docs](https://docs.snowflake.com/en/sql-reference/sql/alter-resource-monitor) you must put something (not triggers or notify users) in the SET clause. The following SQL will work:

```
ALTER RESOURCE MONITOR "some_monitor" 
SET  CREDIT_QUOTA=10
TRIGGERS ON 100 PERCENT DO SUSPEND 
    ON 105 PERCENT DO SUSPEND_IMMEDIATE 
    ON 90 PERCENT DO NOTIFY
```

edit: The following will also work

```
ALTER RESOURCE MONITOR "some_monitor" 
TRIGGERS ON 100 PERCENT DO SUSPEND 
    ON 105 PERCENT DO SUSPEND_IMMEDIATE 
    ON 90 PERCENT DO NOTIFY
```

I would expect the provider to produce this SQL

**Code samples and commands**

Code samples are in the description already

**Additional context**

I assume the same issue exists for changing only the notify users but have not tested this.

**Comment by sfc-gh-jcieslak** [comment]
Hey 👋 
Please refer to this issue - https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1500#issuecomment-2331451228. I'll let you know when the updated resource is available.

**Comment by sfc-gh-jcieslak** [comment]
Hey 👋
The new and refactored resource monitor was released yesterday in version 0.96.0 of the provider. Please migrate with [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0950--v0960) and let me know if the issue has been resolved so we could close the ticket. Thanks in advance 🙏.

**Comment by sfc-gh-jcieslak** [comment]
Closing due to long inactivity. Please, create another issue if you think the problem is still not resolved.
